common/xdc/event_generator.go

```diff
@@ -196,11 +186,54 @@ func (g *EventGenerator) ListResetPoint() []ResetPoint {
 	return g.resetPoints
 }
 
-// SetCanDoBatchOnNextVertex sets a function to determine next generated batch of history events
-func (g *EventGenerator) SetCanDoBatchOnNextVertex(canDoBatchFunc func([]Vertex) bool) {
+// RandomReset randomly pick a reset point and reset the event generator to the point
+func (g *EventGenerator) RandomReset() int {
```

Comment:
RendomizedResetToPointInTime? something like that

Reply:
or RendomResetToResetPoint

Reply:
pick RendomResetToResetPoint